scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/SynchronizationWrapper.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.synchronization;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import org.ow2.proactive.scheduler.common.task.TaskId;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 11/04/2018
+ */
+public class SynchronizationWrapper implements Synchronization {
+
+    String originator;
+
+    TaskId taskId;
+
+    SynchronizationInternal internalAPI;
+
+    public SynchronizationWrapper(String originator, TaskId taskId, SynchronizationInternal internalAPI) {
+        this.originator = originator;
+        this.taskId = taskId;
+        this.internalAPI = internalAPI;
+    }
+
+    @Override
+    public <T extends Serializable> boolean createChannel(String name, boolean isPersistent) throws IOException {
+        return internalAPI.createChannel(originator, taskId, name, isPersistent);
+    }
+
+    @Override
+    public <T extends Serializable> boolean createChannelIfAbsent(String name, boolean isPersistent)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') T is not used in the method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2326)